src/libsyncengine/update_detection/update_detector/node.cpp

```diff
@@ -91,6 +91,21 @@ void Node::setName(const SyncName &name) {
     _name = Utility::normalizedSyncName(name);
 }
 
+bool Node::setParentNode(const std::shared_ptr<Node> &parentNode) {
```

Comment:
We should check that this method is also called in `Node` constructors which are provided with a parent node.

Reply:
For now, I have only just removed the Node constructor that allow to create a Node without parent. Let's re discuss this modification

Reply:
Edit: Finally, I rewrite the `isParentValid` method to climb up the tree and testing only direct parents, instead of going down the tree and testing every child